### PR TITLE
Reland "store ContextState and ModuleMap in embedder slots"

### PR DESCRIPTION
**Pull request by littledivy**
Verified that `node::unit_vm` tests pass in Deno

### CODE REVIEW

File: core/runtime/jsrealm.rs [hunk 1]
```diff
@@ -186,24 +189,42 @@ impl JsRealmInner {
...
+      }
+      ctx.clear_all_slots(isolate);
       // Expect that this context is dead (we only check this in debug mode)
       // TODO(bartlomieju): This check fails for some tests, will need to fix this
       // debug_assert_eq!(Rc::strong_count(&module_map), 1, "ModuleMap still in use.");
     }
-
-    // Expect that this context is dead (we only check this in debug mode)
-    // TODO(mmastrac): This check fails for some tests, will need to fix this
-    // debug_assert_eq!(Rc::strong_count(&self.context), 1, "Realm was still alive when we wanted to destroy it. Not dropped?");
```
Comment (bartlomieju) [inline core/runtime/jsrealm.rs:203]: Why remove this comment?

File: core/runtime/jsruntime.rs [hunk 2]
```diff
@@ -869,7 +869,15 @@ impl JsRuntime {
       );
     }
 
-    context.set_slot(scope, context_state.clone());
+    // SAFETY: We need to initialize the slot. rusty_v8 currently segfaults
+    // when call `clear_all_slots`.
+    unsafe {
+      context.set_slot(scope, ());
```
Comment (bartlomieju) [inline core/runtime/jsruntime.rs:875]: Can we address this in rusty_v8 before landing this PR?

File: core/runtime/jsrealm.rs (hunk 1 above)
Comment (littledivy) [reply core/runtime/jsrealm.rs]: Oops, missed during rebase

File: core/runtime/jsruntime.rs (hunk 2 above)
Comment (littledivy) [reply core/runtime/jsruntime.rs:875]: https://github.com/denoland/rusty_v8/pull/1497

Comment (littledivy) [reply core/runtime/jsruntime.rs]: Fixed

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**